lib/resources/forms.js

```diff
@@ -162,7 +162,7 @@ module.exports = (service, endpoint) => {
           .then(() => Forms.getByProjectAndXmlFormId(params.projectId, params.id, false, Form.DraftVersion))
           .then(getOrNotFound)
         : resolve(form)))
```

Comment:
In the case of publishing a draft form with draft submissions... those submissions get soft-deleted but they _could_ be undeleted later. Exclude draft submissions from `Submissions.getDeleted` to prevent this from happening.